app/controllers/mentee_profiles_controller.rb

```diff
@@ -1,6 +1,38 @@
 class MenteeProfilesController < ApplicationController
-	def show
-	end
-	def dashboard
-	end
+  before_action :require_mentee, except: [ :show ]
+  before_action :require_user, only: [:show]
+
+  def show
+    @user = User.find(params[:id])
+    @mentor = User.find(@user.matched_id)
+  end
+  
+  def dashboard
+    @mentor = User.find(current_user.matched_id)
+  end
+  
+  def missing_mentor
+    @user = current_user
+    @mentor = User.find(current_user.matched_id)
+    if @mentor.update_attribute(:is_missing, true)
+      MissingPersonsMailer.missing_mentor(@user).deliver_now
+      redirect_to dashboard_mentee_profiles_path, notice: 'Organsiers Have been notified about the missing mentor'
+    else 
+      redirect_to dashboard_mentee_profiles_path, notice: 'Couldnt send mail'
+    end
+  end
+  
+  private
+   def require_mentee
```

Comment:
move this method to application_controller as we might want to use is somewhere else later